### ISSUE TITLE
Unable to build with golang 1.10rc1

### ISSUE THREAD

**Issue by soltysh** [apcera/gssapi#54]
We use this library in OpenShift which I recently tried building with `go1.10rc1` and that failed with:
```
# github.com/openshift/origin/vendor/github.com/apcera/gssapi
vendor/github.com/apcera/gssapi/name.go:214: cannot use n.C_gss_name_t (type _Ctype_gss_name_t) as type _Ctype_gss_const_name_t in argument to func literal
```
Any suggestions how this could be fixed?

**Comment by philpennock** [comment]
That looks like you're using very old library headers which are missing some `const`-ification of newer headers, so the fix is to use new enough library headers, per the Building section of the README.

**Comment by liggitt** [comment]
No, it's definitely golang version specific. The package builds fine with go 1.9

**Comment by soltysh** [comment]
Additionally I'd expect Fedora 27 having reasonably recent header files.

**Comment by liggitt** [comment]
it appears that argument coercion is not working in go1.10rc1. reported at https://github.com/golang/go/issues/23720, but in the meantime, an explicit cast would fix this.

**Comment by liggitt** [comment]
Actually, look like golang is fixing the issue prior to 1.10 release... up to you whether you want https://github.com/apcera/gssapi/pull/55 or not. Will leave this open until the golang issue is fixed/verified

**Comment by liggitt** [comment]
fixed in go 1.10rc2

**Comment by soltysh** [comment]
Yup, just verified.
